hkmc2/shared/src/main/scala/hkmc2/codegen/Lowering.scala

```diff
@@ -947,11 +965,17 @@ class Lowering()(using Config, TL, Raise, State, Ctx):
             fsr ::= RcdArg(S(ir), tr)
             rec(as)
     val b = rec(as)
-    val args = if fsr.isEmpty then asr else Arg(N, Value.Rcd(mut = false, fsr.reverse)) :: asr
-    Begin(
-      b,
-      k(args.reverse)
-    )
+    if fsr.isEmpty then
+      Begin(b, k(asr.reverse))
+    else
+      val arrSym = new TempSymbol(N, "rcd")
```

Comment:
Shouldn't this be
```suggestion
      val rcdSym = new TempSymbol(N, "rcd")
```
?

Reply:
Ah, thanks for the catch!

Reply:
Fixed in 6075d309a1dd7914223337f28358a4f3da026f05